[SR SQL Planner Summer Camp] Support anltr4 for ShowProcesslistStmt

predator4ann opened the pull request:
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
Fixes #7960 

## Problem Summary(Required) ：
Support anltr4 for ShowProcesslistStmt

predator4ann committed:
feat: Support anltr4 for ShowProcesslistStmt

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=8487) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=8487) before we can accept your contribution.<br/><hr/>**zacsun** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=8487) it.</sub>

predator4ann commented:
run starrocks_fe_unittest

predator4ann commented:
run starrocks_fe_unittest

imay commented on fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java:
```diff
@@ -0,0 +1,63 @@
+// This file is made available under Elastic License 2.0.
```
new files should use starrocks license header
// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.

predator4ann replied:
done

imay commented:
@predator4ann you can rebase main branch to make fe unittest run success

predator4ann commented:
> @predator4ann you can rebase main branch to make fe unittest run success

@imay I rebase main branch already and fe unittest still failed

predator4ann commented:
@imay thanks for reviewing, I have found the failed reason that there are some fixed time("storage_cooldown_time", "2022-07-09 12:12:12") in unittest

predator4ann commented:
@hiliuxg

wanpengfei-git commented:
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|

HangyuanLiu commented on fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java:
```diff
@@ -0,0 +1,63 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+// This file is based on code available under the Apache license here:
```
delete Apache License

predator4ann replied:
done

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

adzfolc committed:
Revert "[Feature]Support hudi connector (#8356)"

This reverts commit d6d88af4a17f64f63323b1d18cbac2fadd97ca3d.

dirtysalt committed:
[Enhancement] make `io_tasks_per_scan_op` configurable (#8585)

liuyehcf committed:
[Enhancement] Keep trailing zero if specified (#8479)

meegoo committed:
[BugFix] Fix no push_chunk() before set_finishing() in OlapTableSinkOperator (#8274)

#8239 will block pipeline engine in set_finishing(), we need check is_open_done() before open_wait()

meegoo committed:
[Refactor] Refactor transaction stream load interface (#8344)

Currently, we only support single-DB and single-table transactions. In the future, when we support multi-table transactions across DBs, we can keep the interface compatible, and users do not need to modify the URL.

BeginTrasaction:
POST /api/transaction/begin

CommitTransaction:
POST /api/transaction/commit

RollbackTransaction:
POST /api/transaction/rollback

LoadData:
PUT /api/transaction/load

murphyatwork committed:
[Feature] display resource group info in explain verbose (#8481)

Use `explain verbose` to display resource group information. And refactor some chaos code BTW.

E.g.

```sql
MySQL root@127.1:hehe> explain verbose select count(*) from t0;
+--------------------------------------------------------------------------------------------------+
| Explain String                                                                                   |
+--------------------------------------------------------------------------------------------------+
| RESOURCE GROUP: rg_mofei                                                                         |
|                                                                                                  |
| PLAN FRAGMENT 0(F00)                                                                             |
|   Output Exprs:3: count                                                                          |
|   Input Partition: RANDOM                                                                        |
|   RESULT SINK                                                                                    |
|                                                                                                  |
|   1:AGGREGATE (update finalize)                                                                  |
|   |  aggregate: count[(*); args: ; result: BIGINT; args nullable: false; result nullable: false] |
|   |  cardinality: 1                                                                              |
|   |                                                                                              |
|   0:OlapScanNode                                                                                 |
|      table: t0, rollup: t0                                                                       |
|      preAggregation: on                                                                          |
|      partitionsRatio=1/1, tabletsRatio=1/1                                                       |
|      tabletList=41030                                                                            |
|      actualRows=1, avgRowSize=1.0                                                                |
|      cardinality: 1                                                                              |
+--------------------------------------------------------------------------------------------------+
```

Linkerist committed:
reduce lock: locking once and reference it, not locking twice. (#8607)

one lock:
https://github.com/StarRocks/starrocks/blob/f674d92dca0e9ac06cfd8c126d084601c6ab9509/be/src/storage/tablet_updates.cpp#L1047-L1090
one more lock
https://github.com/StarRocks/starrocks/blob/f674d92dca0e9ac06cfd8c126d084601c6ab9509/be/src/storage/tablet_updates.cpp#L2168-L2170
can just lock once.

stdpain committed:
[Enhancement] Support multi-column global dictionary optimization (#7939)

1. If the input column is only a dictionary column and there are no unsupported expressions in this expression, then it must be able to use dictionary optimization
eg: FunctionCall(c1) -> DictMappingExpr(c1)

2. If the input column is multi-column, and if there are expressions in the path of the dictionary column that can use dictionary optimization, then it is also able to use dictionary optimization

eg: FunctionCall(c1, c2, c2) -> FunctionaCall(DictMappingExpr(c1), c2, c2)

but we won't apply this optimize for invalid optimization:
because there is no performance gain .
eg:   concat(a, b) to concat(DictExpr(c), DictExpr(d)).

Linkerist committed:
[Refactor] Add error message in log, when schema change call add_chunk. (#8246)

waittttting committed:
Delete usless code (#8630)

Pslydhh committed:
[BugFix] Check invalid window argument (#8300)

xuzifu666 committed:
[Feature] New Parser and Analyzer support SetUserPropertyStmt/ShowUserPropertyStmt (#8603)

nshangyiming committed:
[BugFix] Fix rate limit of tablet deletion (#8602)

If FE found that meta already deleted when handling tablet report(i.e. tabletMeta == null), we should also control the rate limit of tablet deletion task sent to BE.

dirtysalt committed:
fix reading zero size stream in orc format (#8618)

satanson committed:
fixup avg(distinct non-decimal-type) bugs (#8637)

select avg(distinct non-decimal-type(i.e.BIGINT)) from table gives a invalid multi_distinct_sum signature.
expected: multi_distinct_count[([multi_distinct_count, VARCHAR, false]); args: INT; result: BIGINT
unexpected: multi_distinct_count[([multi_distinct_count, VARCHAR, false]); args: INT; result: DECIMAL128(38,0).

Crystal-LiuJing committed:
[MaterializedView] Generate MaterializedView Full DDL SQL for ShowCreateTableStmt and ShowMaterializedViewStatement  (#8398)

show full ddl_sql for new MaterializedView

HangyuanLiu committed:
[Feature] Add result information for analyze stmt and add general filter code in ShowExecutor #8629

caneGuy committed:
[Feature] refresh materialized view when base tables are refreshed (#7877)

refresh materialized view when base tables are refreshed

liuyehcf committed:
[Enhancement] Slide window support removable cumulative algo (#8241)

xlwh committed:
Cleanup unused AlterClusterClause (#8657)

zhuxt2015 committed:
fix (#8640)

xlwh committed:
[enhancement] Cleanup the old parser for 'AdminSetConfig' (#8659)

wuxueyang96 committed:
[SR SQL Planner Summer Camp] Support ShowStatusStmt for new parser. (#8648)

Commit:
Add Docs (#8617)

dirtysalt committed:
[Enhancement] parquet coalsece reads support across row groups (#8636)

To support coalesce reads across row groups
- allocate `SharedBufferedInputStream` in file reader
- call `release_to_offset` when group reader closed to release memory ASAP

---

Benchmark on SSB-100G, 3 BE nodes, 1 FE nodes. Coalesce-Reads Across Row group does not hurt performance.

Parquet in Zlib

Query | SR(P=8) | Coalesce Reads Across RG | Ratio
-- | -- | -- | --
Q01 | 1574 | 1592 | 1.01
Q02 | 1507 | 1501 | 1
Q03 | 1504 | 1724 | 1.15
Q04 | 2832 | 2693 | 0.95
Q05 | 2611 | 2626 | 1.01
Q06 | 2586 | 3286 | 1.27
Q07 | 3407 | 3465 | 1.02
Q08 | 2834 | 3072 | 1.08
Q09 | 2641 | 2871 | 1.09
Q10 | 2571 | 2555 | 0.99
Q11 | 4499 | 4583 | 1.02
Q12 | 3894 | 3887 | 1
Q13 | 3696 | 3644 | 0.99
SUM | 36156 | 37499 | 1.04

leoyy0316 committed:
[Feature] New Parser and Analyzer support ShowProcStmt (#8621)

imay committed:
[Refactor] Remove Cluster in Load Balancer (#8632)

xlwh committed:
Support show alter in new parser (#8611)

xiaoyong-z committed:
[BugFix] memory-scratch-sink output order error (#8578)

In current implementation, the output of memory order is based on the tuple_descriptors.
This is not corrent, the right order should base on the output_expr.

abc982627271 committed:
[Lake] Support drop lake table  (#8497)

drop lake table:
1.use a daemon thread to drop lake tablet and delete shard
2.persist shard infos in image and journal

Youngwb committed:
[BugFix] Fix unknown error when having clause has subquery (#8662)

ZiheLiu committed:
[Enhancment] Add MorselsCount and TabletCount to profile of scan operator (#8644)

ZiheLiu committed:
[Bugfix] use different OlapScanContexts for non-shared morsel queue (#8643)

ZiheLiu committed:
[Enhancment] Make error message of workgroup more clearly (#8663)

predator4ann committed:
Merge branch 'main' into feature/show_processlist

predator4ann committed:
Update ShowProcesslistStmtTest.java

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
